### ISSUE TITLE
Importing from Medline is done twice, while one import is aborted

### ISSUE THREAD

**Issue by bernhard-kleine** [JabRef/jabref#4841]
JabRef 5.0-dev--snapshot--2019-04-03--master--a273eb1fa
Windows 7 6.1 amd64 
Java 1.8.0_201
- [+] I have tested the latest development version from http://builds.jabref.org/master/ and the problem persists

When importing from Medline, there is the import and one canceled import. Already in version from 2019-03-30. It does not put a strain on the actual import. 
1. use the Pmid of a medline entry e.g. 30137510 to search medline
2. import the article
3. you will notice the importing, but also a message "import canceled"

<details>
  <summary>Log File</summary>
  
  ```
  Paste an excerpt of your log file here
  ```
</details>

**Comment by Siedlerchr** [comment]
Confirmed but nothing important. The jdk calls close on the dialog and this has the same effect as if the cancel button is pressed or the X is pressed and we display the message.
@tobiasdiez  Any elegant solution to handle this?
```

   setResultConverter(button -> {
            if (button.equals(importButton)) {
                viewModel.importEntries(entriesListView.getCheckModel().getCheckedItems());
            } else {
                dialogService.notify(Localization.lang("Import canceled"));
            }

            return null;
        });
```

**Comment by tobiasdiez** [comment]
result converter should be called only once. Can you point me to where the `close` method is called?

**Comment by Siedlerchr** [comment]
the first time when I click import, the result converter is called with OK_DONE. and then in the jdk the close method is called: in the Dialog.class in javafx.scence.control.Dialog (jdk 202)

First call:
```
Thread [JavaFX Application Thread] (Suspended (breakpoint at line 71 in ImportEntriesDialog))	
	ImportEntriesDialog.lambda$0(ButtonType) line: 71	
	844582534.call(Object) line: not available	
	ImportEntriesDialog(Dialog<R>).impl_setResultAndClose(ButtonType, boolean) line: 1026	
	DialogPane.lambda$createButton$599(ButtonType, ActionEvent) line: 777	
	996269360.handle(Event) line: not available	
	CompositeEventHandler$NormalEventHandlerRecord<T>.handleBubblingEvent(T) line: 218	
	CompositeEventHandler<T>.dispatchBubblingEvent(Event) line: 80	
	EventHandlerManager.dispatchBubblingEvent(EventType<Event>, Event) line: 238	
	EventHandlerManager.dispatchBubblingEvent(Event) line: 191	
	NodeEventDispatcher(CompositeEventDispatcher).dispatchBubblingEvent(Event) line: 59	
	NodeEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 58	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	NodeEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	SceneEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	WindowEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	EventUtil.fireEventImpl(EventDispatchChain, EventTarget, Event) line: 74	
	EventUtil.fireEvent(EventTarget, Event) line: 49	
	Event.fireEvent(EventTarget, Event) line: 198	
	Button(Node).fireEvent(Event) line: 8411	
	Button.fire() line: 185	
	ButtonBehavior<C>.mouseReleased(MouseEvent) line: 182	
```
Second call:
```
	ImportEntriesDialog(Dialog<R>).impl_setResultAndClose(ButtonType, boolean) line: 1026	
	ImportEntriesDialog(Dialog<R>).close() line: 398 [local variables unavailable]	
	ImportEntriesDialog(Dialog<R>).impl_setResultAndClose(ButtonType, boolean) line: 1036	
	DialogPane.lambda$createButton$599(ButtonType, ActionEvent) line: 777	
	996269360.handle(Event) line: not available	
	CompositeEventHandler$NormalEventHandlerRecord<T>.handleBubblingEvent(T) line: 218	
	CompositeEventHandler<T>.dispatchBubblingEvent(Event) line: 80	
	EventHandlerManager.dispatchBubblingEvent(EventType<Event>, Event) line: 238	
	EventHandlerManager.dispatchBubblingEvent(Event) line: 191	
	NodeEventDispatcher(CompositeEventDispatcher).dispatchBubblingEvent(Event) line: 59	
	NodeEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 58	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	NodeEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	SceneEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	WindowEventDispatcher(BasicEventDispatcher).dispatchEvent(Event, EventDispatchChain) line: 56	
	EventDispatchChainImpl.dispatchEvent(Event) line: 114	
	EventUtil.fireEventImpl(EventDispatchChain, EventTarget, Event) line: 74	
	EventUtil.fireEvent(EventTarget, Event) line: 49	
	Event.fireEvent(EventTarget, Event) line: 198	
	Button(Node).fireEvent(Event) line: 8411	
	Button.fire() line: 185	
	ButtonBehavior<C>.mouseReleased(MouseEvent) line: 182	
```

**Comment by tobiasdiez** [comment]
Seems to be a bug in JavaFX: https://stackoverflow.com/questions/48267935/cannot-return-null-in-javafx-dialog-result-converter

Probably easiest fix is to change the return type from `Void` to `Boolean` and return false instead of `null`. This might affect also more dialogs than just this one.

**Comment by Siedlerchr** [comment]
Fixed in #6269
